shape_msgs/SolidPrimitive[] primitives
geometry_msgs/Pose[] poses